File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
                int N=prices.length;
        if(N==0)return 0;
        int  maxProfit=0;
        int minSofar=prices[0];
        for (int a : prices) {
            maxProfit=Math.max(maxProfit,a-minSofar);
            minSofar=Math.min(minSofar,a);
        }

        return maxProfit;
    }
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l,r=0,1
        maxProfit=0
        N=len(prices)
        while r<N:
            if prices[r]>prices[l]:
                maxProfit=max(maxProfit,prices[r]-prices[l])
            else:
                l=r
            r+=1
        return maxProfit
                
                
                
            
            
        
        